components/sup/src/manager/service/mod.rs

```diff
@@ -279,64 +264,67 @@ impl Service {
     pub fn tick(&mut self, census_ring: &CensusRing) -> bool {
         if !self.initialized {
             if !self.all_bindings_present(census_ring) {
-                outputln!(preamble self.service_group, "Waiting to initialize service.");
+                outputln!(preamble self.service_group, "Waiting for service binds...");
                 return false;
             }
         }
-        let svc_cfg_updated = self.update_configuration(census_ring);
+
+        let svc_updated = self.update_templates(census_ring);
+        if self.update_service_files(census_ring) {
+            self.file_updated();
+        }
 
         match self.topology {
             Topology::Standalone => {
-                self.execute_hooks();
+                self.execute_hooks(census_ring);
```

Comment:
I follow your reasoning but I'm talking about having 2 explicit phases.
* render config
* execute hooks

I can't see why this should have anything to do with the language.